src/App/Fossa/Config/ConfigFile.hs

```diff
@@ -259,9 +260,17 @@ instance FromJSON ConfigProject where
       <*> obj .:? "team"
       <*> obj .:? "jiraProjectKey"
       <*> obj .:? "url"
-      <*> obj .:? "policy"
+      <*> parsePolicy obj
       <*> obj .:? "labels" .!= []
       <*> obj .:? "releaseGroup"
+      <*> obj .:? "policyId"
+    where
+      parsePolicy obj = do
+        pName <- obj .:? "policy"
+        pId <- obj .:? "policyId"
+        case (pName, pId) of
+          (Just _, Just _) -> fail "Only one of 'policy' or 'policyId' can be set at a time."
+          _ -> pure $ (PolicyName <$> pName) <|> (PolicyId <$> pId)
```

Comment:
I believe this functionality is different than when provided in the CLI. Please unify the functionality such that FOSSA CLI always either chooses one, or rejects if both are provided.

Reply:
Can you talk a bit more about the differences you see? As far as I can tell the CLI does the same thing. Note that for `optparse-applicative` `<|>` combines option parsers in a pick-only-one kind of way which I wasn't expecting initially. Not sure if this is what you were referring to.

Reply:
When I run `cabal run fossa -- analyze --policy hello --policy-id 123` I get an error as expected. It's an ugly error, I couldn't see a way to make it nice, but it does fail.

Are you thinking of this [line](https://github.com/fossas/fossa-cli/pull/1203/files#diff-81606c0dad857cde73069f8f95ee8661c45d303c5a2c0e329eedbf3f26439a90R171)?
```haskell
parsePolicyOptions = optional (policy <|> policyId) -- For Parsers '<|>' tries every alternative and fails if they all succeed.
```
`optparse-applicative`s implementation of `<|>` for `Parser`s accepts the first alternative to succeed, unless more than one is included. Docs [here](https://github.com/pcapriotti/optparse-applicative#alternative). 

I may be missing a case though.

Reply:
Oh, I misunderstood how this worked! My mistake, thanks for clarifying. In that case, I think we're good to go.

Reply:
Thanks! I spent an embarrassing amount of time when implementing this to get that behavior because it didn't ever occur to me that the answer would just be to use `<|>`. It's definitely understandable how it'd be confusing.